server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java

```diff
@@ -59,6 +59,13 @@ private ReindexRequest(SearchRequest search, IndexRequest destination, boolean s
         this.destination = destination;
     }
 
+    public ReindexRequest(StreamInput in) throws IOException {
+        super.readFrom(in);
```

Comment:
@nik9000 Is this alright or should we do it also for the whole chain?

Reply:
If doing the whole chain makes the PR too big, I'd do this. IT should be fine. If the PR isn't too big then I'd do the whole chain.

Reply:
It does make it kind of complicated cause the chain is rather long. For now, I will leave it like this.

Reply:
I was going to comment on this, but I understand how much this blows things up. So punting it sounds good!